pkg/operation/botanist/secrets.go

```diff
@@ -306,6 +307,32 @@ func (b *Botanist) syncShootCredentialToGarden(
 	return err
 }
 
+func (b *Botanist) deleteSSHKeypairGardenSecret(ctx context.Context) error {
+	if err := b.deleteGardenSecret(ctx, gutil.ShootProjectSecretSuffixSSHKeypair); err != nil {
+		return err
+	}
+
+	if err := b.deleteGardenSecret(ctx, gutil.ShootProjectSecretSuffixOldSSHKeypair); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (b *Botanist) deleteGardenSecret(ctx context.Context, nameSuffix string) error {
```

Comment:
I suggest that we define this function in the `deleteSSHKeypair` function or we change its name to something that is not so generic. Right now I cannot understand what this function does without looking in its implementation. Maybe similarly to `syncShootCredentialToGarden` we can call it `deleteShootCredentialGarden`.

We should also have a test case to cover this deletion. See [this](https://github.com/gardener/gardener/blob/0c9b1b4924e1757ebb7d3a174f88b0f9de8d6684/pkg/operation/botanist/secrets_test.go#L134) test as example of the opposite.